crates/esplora/src/blocking_ext.rs

```diff
@@ -250,7 +250,8 @@ impl EsploraExt for esplora_client::BlockingClient {
                     }
                 }
 
-                if last_index > last_active_index.map(|i| i.saturating_add(stop_gap as u32)) {
+                let last_index = last_index.expect("Must be set since handles wasn't empty.");
+                if last_index > last_active_index.unwrap_or(0) + stop_gap as u32 {
```

Comment:
```suggestion
                if last_index > last_active_index.unwrap_or(0).saturating_add(stop_gap as u32) {
```
Since the version without keychain set the gap limit to uint32::MAX.